include/io.h

```diff
@@ -14,40 +18,61 @@
 class IO {
  public:
   //! \brief Make constructor for input and output files
-  IO(const std::string& pathfile_name) {
-    
-    //! Open path file
-    std::ifstream pathfile(pathfile_name);
-    pathfile.exceptions(std::ifstream::badbit);
-
-    //! Read file and store to private variables
-    nlohmann::json j;
-    pathfile >> j;
-    inputfilename_ = j["inputfile"].get<std::string>();
-    outputfilename_point_ = j["outputfile_point"].get<std::string>();
-    outputfilename_stress_ = j["outputfile_stress"].get<std::string>();
+  //! Explicit to protect against implicit casting
+  explicit IO(const std::string& pathfile_name) {
+
+    try {
+      //! Open path file
+      std::ifstream pathfile(pathfile_name);
+      pathfile.exceptions(std::ifstream::badbit);
+
+      //! Read file and store to private variables
+      path_input_ = nlohmann::json::parse(pathfile);
+      inputfilename_ = path_input_["inputfile"].get<std::string>();
+      outputfilename_vertex_ =
+          path_input_["outputfile_vertex"].get<std::string>();
+      outputfilename_stress_ =
+          path_input_["outputfile_stress"].get<std::string>();
+
+    } catch (std::exception& except) {
+      std::cout << "Caught exception: " << except.what() << '\n'
+                << "Please check the path file.\n";
+    }
   }
 
-  //! \brief Get the private properties
+  //! \brief Write output file for point
+  void write_output_vertices(
+      const std::string& outputfilename,
+      const std::vector<std::shared_ptr<Point<3>>> vertices,
+      const unsigned& tot_points);
```

Comment:
This should be pass by value not `const&`. See: https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rf-in